frontend/src/routes.js

```diff
@@ -67,59 +66,57 @@ function SidebarLink({ to, children, ...rest }) {
   );
 }
 
-const DefaultLayout = ({ children }) => (
-  <Fragment>
-    <div
-      className={css({
-        height: '100%',
-        minHeight: '100%',
-        overflow: 'hidden',
-        backgroundColor: '#f5f7fa',
-      })}
-    >
-      <Header />
-      <Container>
-        <Sidebar>
-          <SidebarLink className={css({ marginTop: 30 })} to="/inbox">
-            Inbox
-          </SidebarLink>
-          <SidebarLink to="/accounts">Accounts</SidebarLink>
-          <SidebarLink to="/contacts">Contacts</SidebarLink>
-          <SidebarLink to="/web">Web</SidebarLink>
-          <SidebarLink to="/files">Files</SidebarLink>
-        </Sidebar>
-        <div className={css({ flex: 1 })}>{children}</div>
-      </Container>
-    </div>
-  </Fragment>
+const DefaultLayout = ({ children, renderProps }) => (
+  <div
+    className={css({
+      height: '100%',
+      minHeight: '100%',
+      overflow: 'hidden',
+      backgroundColor: '#f5f7fa',
+    })}
+  >
+    <Header renderProps={renderProps} />
+    <Container>
+      <Sidebar>
+        <SidebarLink className={css({ marginTop: 30 })} to="/inbox">
+          Inbox
+        </SidebarLink>
+        <SidebarLink to="/contacts">Contacts</SidebarLink>
+        <SidebarLink to="/accounts">Accounts</SidebarLink>
+        <SidebarLink to="/web">Web</SidebarLink>
+        <SidebarLink to="/files">Files</SidebarLink>
+      </Sidebar>
+      <div className={css({ flex: 1 })}>{children}</div>
+    </Container>
+  </div>
 );
 
-const loaderProps = {
-  active: true,
-  withOverlay: true,
-  small: false,
-};
-
 const Welcome = () => (
   <WelcomeContainer>
     <h1>Welcome to PhishTray</h1>
   </WelcomeContainer>
 );
 
-const Entry = param => (
-  <div>
-    <Loading
-      {...loaderProps}
-      className={css({
-        stroke: '#161415!important',
-      })}
-    />
-    {param ? (
-      <Redirect from="/" to="/inbox" />
-    ) : (
-      <Redirect from="/" to="/welcome" />
-    )}
-  </div>
+const PrivateRoute = ({ component: Component, isAllowed, ...rest }) => (
+  <Route
+    {...rest}
+    render={props => {
+      if (isAllowed)
+        document.title = `Phishtray | ${getHeaderText(props.match.path)}`;
+      return isAllowed ? (
+        <DefaultLayout renderProps={props}>
+          <Component {...props} />
+        </DefaultLayout>
+      ) : (
+        <Redirect
+          to={{
+            pathname: '/welcome',
+            state: { from: props.location },
+          }}
+        />
```

Comment:
is it possible to do:

    if (isAllowed) {
        document.title = ...
        return (<DefaultLayout ...> </DefaultLayout>)
    else:
        return (<Redirect ... />)